### PR TITLE
BOXJS Integration with cuckoo

### PR DESCRIPTION
**Pull request by Nwinternights**
Thanks for contributing! But first: did you read our community guidelines?
https://cuckoo.sh/docs/introduction/community.html

##### What I have added/changed is:
New Processing module that  uses boxjs as an advanced Js deobfuscator and evaluator. It runs in a docker env with a small Rest Service(based on NodeJS).  https://github.com/CapacitorSet/box-js.
##### The goal of my change is:
Provides more detection of JS files. The how-to to set up a Docker Boxjs is here https://github.com/CapacitorSet/box-js/tree/master/integrations 
##### What I have tested about my change is:
boxjs
boxjs2
tks to CapacitorSet and frank2411

### CODE REVIEW

File: cuckoo/processing/boxjs.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        self.key = "boxjs"
+
+        """ Fall off if we don't deal with files """
+        if self.results.get("info", {}).get("category") != "file":
+            log.debug("Box-js supports only file scanning!")
+            return {}
+
+        self.url = self.options.get("url")
+        self.timeout = int(self.options.get("timeout", 60))
+	self.ioc = self.options.get("IOC")
```
Comment (doomedraven) [inline cuckoo/processing/boxjs.py]: fix ;) alineation

File: cuckoo/processing/boxjs.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            if retry:
+                postUrl = urlparse.urljoin(self.url, "/sample")
+                analysis_id = self._post_text(postUrl,  sample=open(self.file_path, "rb"), flags=flags)# returns a UUID
+                base_url = "{}/sample/{}".format(self.url, str(analysis_id))
+
+        # Fetch the results.
+        results = {}
+        urls_url = "{}/urls".format(base_url)
+        resources_url = "{}/resources".format(base_url)
+	iocs_ioc = "{}/IOC".format(base_url)
```
Comment (doomedraven) [inline cuckoo/processing/boxjs.py]: same

File: cuckoo/processing/boxjs.py (hunk 1 above)
Comment (doomedraven) [reply cuckoo/processing/boxjs.py]: there few more in code

File: cuckoo/processing/boxjs.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            if retry:
+                postUrl = urlparse.urljoin(self.url, "/sample")
+                analysis_id = self._post_text(postUrl,  sample=open(self.file_path, "rb"), flags=flags)# returns a UUID
+                base_url = "{}/sample/{}".format(self.url, str(analysis_id))
+
+        # Fetch the results.
+        results = {}
+        urls_url = "{}/urls".format(base_url)
+        resources_url = "{}/resources".format(base_url)
+		iocs_ioc = "{}/IOC".format(base_url)
```
Comment (doomedraven) [inline cuckoo/processing/boxjs.py]: broken, and few more

**Comment by Nwinternights** [comment]
https://github.com/Nwinternights/box2cuckoo for a quick setup of Boxjs API Docker
@doomedraven let me know if u tried

**Comment by doomedraven** [comment]
@Nwinternights thanks i will try it

**Comment by doomedraven** [comment]
the dll stuff is already in PRs, don't mess with different stuff in the same payload, from my exp it will be harder to get merged

**Comment by Nwinternights** [comment]
u're right , my fault!

**Pull request closed without merging**